wiki/Gameplay/Judgement/osu!/en.md

```diff
@@ -1,57 +1,88 @@
-- 100: The player hit at least half of the slider parts.
-- 50: The player managed to hit at least one of the parts.
-- Miss: The player did not successfully hit any of the slider parts.
-- If [ScoreV2](/wiki/Gameplay/Game_modifier/ScoreV2) is active, the accuracy of hitting the slider head is additionally accounted for when judging the entire slider:
-  - Getting a 300 for a slider requires getting a 300 judgement for the slider head.
-  - Getting a 100 for a slider requires getting a 100 judgement or better for the slider head.
+The slider head only needs to be hit within the MEH hit window for a successful hit. However, if [ScoreV2](/wiki/Gameplay/Game_modifier/ScoreV2) is active, slider head accuracy is additionally accounted for, with the judgement received for the entire slider being limited to at most the one received for the slider head.
 
-The slider parts also have additional influence on the current [combo](/wiki/Gameplay/Combo_(score_multiplier)):
+There are some additional quirks with how sliders influence [combo](/wiki/Gameplay/Combo_(score_multiplier)):
 
-- Tapping the slider head too early does not incur a miss, but will reset combo to 0.
-- Missing a slider tick or repeat does not incur a miss, but will reset combo to 0.
-- Missing the slider end does not incur a miss, but will not increment combo.
+- Tapping the slider head too early (before the MEH hit window), missing a slider tick, or missing a repeat does not incur a MISS, but will cause a [combo break](/wiki/Gameplay/Judgement/Combobreak). The other slider parts can still be hit if a key is held down. This is colloquially referred to as a [slider break](/wiki/Gameplay/Judgement/Slider_break).
+- MISSing the slider end does not incur a MISS, but will not increment combo.
 
 ### Spinners
 
-Each spinner has a set number of spins required to complete it. This number depends on the [overall difficulty](/wiki/Beatmap/Overall_difficulty#sliders-and-spinners) of the beatmap. The criteria for judging the spinner as a whole are as follows:
+Each [spinner](/wiki/Gameplay/Hit_object/Spinner) has a set number of spins required to complete it. This number depends on the [overall difficulty](/wiki/Beatmap/Overall_difficulty#sliders-and-spinners) of the beatmap.
+
+| Judgement | Spins required[^half-spins] |
+| :-: | :-- |
+| GREAT | 100% |
+| OK | One spin less than the required number |
+| MEH | 25% |
+| MISS | 0% |
+
+The spin requirements can be broken down into the following formulas:
 
-- 300: The player successfully performed the number of spins required for completion, or more.
-- 100: The player performed one spin less than the required number.
-- 50: The player performed at least 25% of the required spins.
-- Miss: The player failed to meet the minimum of 25% of spins required.
+|  |  |
+| :-- | :-- |
+| Minimum spins per second | `1.5 + 0.2 * OD` if OD < 5, `1.25 + 0.25 * OD` if OD >= 5 |
+| Minimum spins required | Spinner length in seconds * minimum spins per second + 0.5 |
```

Comment:
after checking with stable source apparently there were some random bits of information that were inconsistent

- the formula for the minimum required spins per second from <https://osu.ppy.sh/wiki/en/Beatmap/Overall_difficulty#sliders-and-spinners> is the exact formula used in the code, except that it deals in half rotations which has been adjusted here
- this random +0.5 relates to the "All spinners required half of a spin more to complete." trivia in the history section. all spinners really were nerfed by 1/2 rotation for some ungodly reason (before it would've been +1)
  - due to this the "minimum spins per second" formula listed here is actually slightly less than what you'd need (by `0.5 / length in seconds` so not much in actuality)
- the "clear" text for spinners actually appears *half of a rotation* before you'd get a GREAT, and *half of a rotation* after you'd get an OK (if you stopped immediately you'd get an OK)
- the "One spin less than the required number" for the OK requirement was in terms of full spins, so that part didn't need changing

> I'd highly recommend not attempting to figure this out from the stable code base.

well the judgement part wasn't so bad to investigate, but figuring out maximum score possible (involving maximum possible rotations) is way more involved. there is a [desmos calculator for this](https://www.desmos.com/calculator/fpkubgscmx) though and it's true that it would be better to empirically test that against stable (but the that info would be somewhere like `Gameplay/Score` or `Gameplay/Hit_object/Spinner`)